### PR TITLE
A handful of low-hanging sca warning fixes

### PR DESCRIPTION
**Pull request by notspiff**
See title.

### CODE REVIEW

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -124,7 +124,7 @@ CFileItem::CFileItem(const CVideoInfoTag& movie)
   SetFromVideoInfoTag(movie);
 }
 
-void CFileItem::FillMusicInfoTag(const CPVRChannelPtr channel, const CPVREpgInfoTagPtr tag)
+void CFileItem::FillMusicInfoTag(const CPVRChannelPtr& channel, const CPVREpgInfoTagPtr& tag)
```
Comment (FernetMenta) [inline xbmc/FileItem.cpp]: passing a smart pointer by ref does not make much sense to me. pass the object it holds by ref or have the shared pointer increment by passing by value. passng a smart pointer by ref can hit you badly in a multithreaded environment

Comment (notspiff) [reply xbmc/FileItem.cpp]: for mutable refs, sure. for const refs, you do not get any additional safety by doing an increase/decrease cycle. it's not being stored anywhere, we just read data and return.

File: xbmc/cores/DataCacheCore.cpp [hunk 2]
```diff
@@ -26,7 +26,8 @@ CDataCacheCore::CDataCacheCore() :
   m_playerVideoInfo {},
   m_playerAudioInfo {},
   m_renderInfo {},
-  m_stateInfo {}
+  m_stateInfo {},
```
Comment (FernetMenta) [inline xbmc/cores/DataCacheCore.cpp]: can we do this in the header? imo much better than having ever growing ctors.

Comment (notspiff) [reply xbmc/cores/DataCacheCore.cpp]: i prefer it. i have tried that earlier and been told no init in in header.

Comment (notspiff) [reply xbmc/cores/DataCacheCore.cpp]: done.

File: xbmc/FileItem.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/FileItem.cpp]: > for mutable refs, sure. for const refs, you do not get any additional safety by doing an increase/decrease cycle.

 If you want to ensure that the object behind the shared pointer is kept alive while the method is called, passing it by value is eactly what you want. As @FernetMenta said, this is generally a good approach in multithreaded environment.

Comment (notspiff) [reply xbmc/FileItem.cpp]: please. read the surrounding code. read what i wrote. i know very well how shared pointers work. if shit manage to remove it from under our feet in this helper function that reads and immediately returns, something else is wrong. you are just stressing the ref counting for absolutely no reason here. but whatever, i will just remove this commit and let you guys act on generics and not logic.

Comment (ksooo) [reply xbmc/FileItem.cpp]: Got it. See my EDIT...

Comment (ksooo) [reply xbmc/FileItem.cpp]: Please readd the commit.

Comment (notspiff) [reply xbmc/FileItem.cpp]: round and round it goes. readded

Comment (ksooo) [reply xbmc/FileItem.cpp]: Thanks. You were just to fast for me rethinking what shit I wrote in the first place. ;-)

File: xbmc/cores/DataCacheCore.cpp (hunk 2 above)
Comment (FernetMenta) [reply xbmc/cores/DataCacheCore.cpp]: thanks

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
all green. Good to go?

**Pull request merged**